sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -57,6 +62,11 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * @since 1.4.0
    */
   def schema(schema: StructType): DataFrameReader = {
+    if (this.userSpecifiedSchema.nonEmpty) {
```

Comment:
i'd probably get rid of this one too -- it is technically behavior changing and we are now taking away the possibility for users to just override a format. Did you actually see somebody getting confused by this?